appinventor/blocklyeditor/src/workspace_svg.js

```diff
@@ -632,9 +665,60 @@ Blockly.WorkspaceSvg.prototype.customContextMenu = function(menuOptions) {
     var comparatorA = comparisonName(a).toLowerCase();
     var comparatorB = comparisonName(b).toLowerCase();
 
-    if (comparatorA < comparatorB) return -1;
-    else if (comparatorA > comparatorB) return +1;
-    else return 0;
+    if (a.category != b.category){
```

Comment:
If you see my earlier comment about how to use localeCompare, I think all of this logic can go away and we let the comparison function order things based on the leading numerals.